lib/location/method_gnss.c

```diff
@@ -866,6 +869,106 @@ static void method_gnss_print_pvt(const struct nrf_modem_gnss_pvt_data_frame *pv
 	}
 }
 
+#if defined(CONFIG_LOCATION_SERVICE_NRF_CLOUD_GNSS_POS_CLOUD_SEND)
+
+static int location_cloud_send(char *body)
+{
+	int ret;
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	struct nrf_cloud_tx_data mqtt_msg = {
+		.data.ptr = body,
+		.data.len = strlen(body),
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE,
+		.topic_type = NRF_CLOUD_TOPIC_MESSAGE,
+	};
+
+	ret = nrf_cloud_send(&mqtt_msg);
+	if (ret) {
+		LOG_ERR("MQTT: location data sending failed: %d", ret);
+	}
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	ret = nrf_cloud_coap_json_message_send(body, false, true);
+	if (ret) {
+		LOG_ERR("CoAP: location data sending failed");
+	}
+#elif defined(CONFIG_NRF_CLOUD_REST)
+#define REST_DETAILS_RX_BUF_SZ 300 /* No payload in a response, "just" headers */
+	static char rx_buf[REST_DETAILS_RX_BUF_SZ];
+	static char device_id[NRF_CLOUD_CLIENT_ID_MAX_LEN + 1];
+	static struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.rx_buf = rx_buf,
+		.rx_buf_len = sizeof(rx_buf),
+		.fragment_size = 0
+	};
+
+	ret = nrf_cloud_client_id_get(device_id, sizeof(device_id));
+	if (ret == 0) {
+		ret = nrf_cloud_rest_send_device_message(&rest_ctx, device_id, body, false, NULL);
+		if (ret) {
+			LOG_ERR("REST: location data sending failed: %d", ret);
+		}
+	} else {
+		LOG_ERR("Failed to get device ID, error: %d", ret);
+	}
+#endif
+	return ret;
+}
+
+void method_gnss_cloud_send(struct location_data *location_result)
+{
+	int err = 0;
+	char *json_str_out = NULL;
+	cJSON *gnss_data_obj = NULL;
+	struct nrf_cloud_gnss_data gnss_data = {
+		.type = NRF_CLOUD_GNSS_TYPE_PVT,
+		.ts_ms = NRF_CLOUD_NO_TIMESTAMP, /* TODO: What should be here? */
```

Comment:
Not sure we want to add dependency to Date Time library to use `date_time_now()`.
location_result.datetime would be available which has been taken from PVT data.
Or then NRF_CLOUD_NO_TIMESTAMP which means cloud will use a time when it receives position data.

Reply:
ATv2 seems to use `k_uptime_get()`.

Reply:
The timestamp could also be calculated from the time in the PVT. That would be the most accurate time available.

Reply:
Yep, time from PVT is possible. I'm wondering why ATv2 hasn't used it but I guess we've just made things "easier" without that conversion.
The easiest would be to use NRF_CLOUD_NO_TIMESTAMP and cloud uses the time from when the data arrives there.

Reply:
Using time from location event which is time from PVT data. `timeutil_timegm64()` used for conversion.

Reply:
I think it would be best to use GNSS time or date_time library time, those should be accurate and because why not. Cloud time is probably also accurate, but it may take some time before the data is received by the cloud.